lib/Table/TableLegendStratum.ts

```diff
@@ -141,12 +141,12 @@ export class TableAutomaticLegendStratum extends LoadableStratum(
     ]);
   }
 
+  @computed get hideLegendInWorkbench() {
+    return (this._item.dataColumnMajor ?? []).length === 0;
+  }
+
   @computed get legends(): StratumFromTraits<LegendTraits>[] {
-    if (
-      this._item.mapItems.length === 0 ||
-      (this._item.dataColumnMajor ?? []).length === 0
-    )
-      return [];
+    if (this._item.mapItems.length === 0) return [];
 
```

Comment:
Instead of returning no legend, we now hide the legend - this is so we can override it in `ArcGisFeatureServerCatalogItem`